### DISCUSSION TITLE
#11 📑 Weekly Report 11

### DISCUSSION THREAD

**Discussion by ins305** [SE-TINF22B6/Statify#87] [Blog]
## UML Diagram
This week we created a new [UML Component Diagram](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/diagrams/UML_Component_Diagram.svg) to give you a better overview on how our project is divided into different elements and how they interact with each other.
![UML_Component_Diagram](https://github.com/SE-TINF22B6/Statify/assets/64202248/e952d4d3-1373-4044-ad5b-93481e4f4cdd)
The three main parts of our project are the Frontend, Backend and the Database. And of course our application also needs to interact with the Spotify Web API.
Frontend and Backend communicate via our Spring Boot REST API through the endpoints defined in the SpotifyController class of our Backend.
With the UserInterface, the Frontend enables our users to view their statistics. To do so, it sends a request to the Backend, where the Statistics component returns either already generated statistics or it generates new ones from the user's streams saved in the database.

## Software Architecture Documentation
We also started with the [Software Architecture Documentation](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/sad.md) for our project. Here you can already read about the Component Diagram and we will continue to add more information soon.

## Backend

I'm pleased to report significant progress with our streaming daemon—it's nearly fully implemented! We've established the thread logic for effective scheduling and integrated methods for retrieving tracks and streaming songs within this environment. Now, just a few tests and tweaks remain. Thank you for your continued support as we finalize these last few steps and let us show you our progress in the backend.

**Comment by LenVej** [comment]
Dear Statify Team,

Your blog provides a clear overview of your project's architecture, with a well-defined UML Component Diagram showcasing the interaction between Frontend, Backend, and Database, as well as integration with the Spotify Web API. The use of a layered architecture is justified, ensuring data flow through distinct layers.
Keep up the good work! 👍

Your CookHub Team

**Comment by derrrobin** [comment]
Hey there, 

Impressive update! The new UML Component Diagram clarifies project structure, while progress on the Software Architecture Documentation shows a commitment to transparency. Backend development progress, especially with the streaming daemon, is notable. Looking forward to seeing more progress!

**Comment by MargotGsb** [comment]
Dear Team Statify,

I think you gave us a very good overview over your general architcture. I am very excited to see your future progress with your streaming daemon. Keep this amazing work up!

Your Team Plapy :D
